src/sema/tests/mod.rs

```diff
@@ -128,3 +139,265 @@ fn test_statement_reachable() {
+        "#;
+    let ns = parse(file);
+    let errors = ns.diagnostics.errors();
+    let warnings = ns.diagnostics.warnings();
+
+    assert_eq!(errors[0].message, "value 133 does not fit into type int8.");
+    assert_eq!(errors[1].message, "negative value -1 does not fit into type uint8. Cannot implicitly convert signed literal to unsigned type.");
+    assert_eq!(errors[2].message, "value 133 does not fit into type int8.");
+    assert_eq!(errors[3].message, "negative value -1 does not fit into type uint8. Cannot implicitly convert signed literal to unsigned type.");
+    assert_eq!(errors[4].message, "value 762 does not fit into type int8.");
+    assert_eq!(errors[5].message, "value 882 does not fit into type int8.");
+    assert_eq!(errors[6].message, "value 128 does not fit into type int8.");
+    assert_eq!(errors[7].message, "value 128 does not fit into type int8.");
+    assert_eq!(errors[8].message, "negative value -1 does not fit into type uint8. Cannot implicitly convert signed literal to unsigned type.");
+    assert_eq!(errors[9].message, "value 129 does not fit into type int8.");
+    assert_eq!(errors[10].message, "negative value -1 does not fit into type uint8. Cannot implicitly convert signed literal to unsigned type.");
+    assert_eq!(errors[11].message, "value 129 does not fit into type int8.");
+    assert_eq!(
+        errors[12].message,
+        "value 4294967296 does not fit into type uint32."
+    );
+    assert_eq!(errors[13].message, "value 130 does not fit into type int8.");
+    assert_eq!(
+        errors[14].message,
+        "value 300 does not fit into type uint8."
+    );
+    assert_eq!(
+        errors[15].message,
+        "value 301 does not fit into type uint8."
+    );
+    assert_eq!(
+        errors[16].message,
+        "value 4294967296 does not fit into type uint32."
+    );
+    assert_eq!(errors[17].message, "value 240 does not fit into type int8.");
+    assert_eq!(errors[18].message, "value 245 does not fit into type int8.");
+    assert_eq!(
+        errors[19].message,
+        "value 260 does not fit into type uint8."
+    );
+    assert_eq!(
+        errors[20].message,
+        "value 261 does not fit into type uint8."
+    );
+    assert_eq!(
+        errors[21].message,
+        "value 269 does not fit into type uint8."
+    );
+    assert_eq!(errors[22].message, "value 155 does not fit into type int8.");
+    assert_eq!(
+        errors[23].message,
+        "value 262 does not fit into type uint8."
+    );
+
+    assert_eq!(
+        errors[24].message,
+        "value 744 does not fit into type uint8."
+    );
+
+    assert_eq!(errors.len(), 25);
+
+    assert_eq!(
+        warnings[1].message,
+        "left shift by 7 may overflow the final result"
+    );
+    assert_eq!(
+        warnings[2].message,
+        "left shift by 7 may overflow the final result"
+    );
+    assert_eq!(
+        warnings[3].message,
+        "left shift by 9 may overflow the final result"
+    );
```

Comment:
How do you make sure there are no other warnings besides these?

Reply:
I added an assert for the number of warnings, thanks